binary_sensor.py

```diff
@@ -114,14 +115,14 @@ def name(self):
     def device_info(self):
         """Return information to link this entity with the correct device."""
         return {
-            "identifiers": {(DOMAIN, self.device_data.device_client.passcode)},
-            "name": self.device_data.device_client.get_unique_name(),
+            "identifiers": {(DOMAIN, self.device_data.passcode())},
+            "name": self.device_data.unique_name(),
```

Comment:
What happens to entities if these are `unknown`?

Reply:
They get duplicated. To prevent this, I've added a `connected()` function that can be used to check if the client is connected. If it isn't, entities do not get registered.